src/styles/index.scss

```diff
@@ -1,3 +1,64 @@
+@import 'variables';
+
+* {
+  box-sizing: border-box;
+}
+
 body {
   margin: 0;
+  height: 100vh;
+  width: 100vw;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap-width;
+  width: calc($width-day * 7 + 10px * 2 + $gap-width * 6);
+  padding: 10px;
+
+  &__day {
+    position: relative;
+    width: $width-day;
+    height: $height-day;
+    background-color: #eee;
+    border: 1px solid black;
+    cursor: pointer;
+    transition:
+      transform 0.5s,
+      background-color 0.5s;
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i})::before {
+        content: '#{$i}';
+        position: absolute;
+        font-size: 30px;
+        color: black;
+        font-family: Arial, Helvetica, sans-serif;
+        top: 50%;
+        left: 50%;
+        transform: translate(-49%, -50%);
+      }
+
+      &:hover {
+        background-color: #ffbfcb;
+        transform: translateY(-20px);
+      }
+    }
+  }
+
+  @each $day, $index in $days {
+    &--start-day-#{$day} &__day:first-child {
```

Comment:
The BEM naming convention suggests that modifiers should be used with the block or element name they modify. In this case, `&--start-day-#{$day}` should be applied to the `.calendar` block, not the `.calendar__day` element, since `start-day` is a modifier of the calendar itself.